Tests/Unit/Event/Decorator/EventWithMetadataTest.php

```diff
@@ -61,6 +63,6 @@ public function eventIsUnwrappedWhenNestingEventsWithMetadata(): void
         $eventWithMetadata = new EventWithMetadata($this->mockEvent, []);
         $nestedEventWithMetadata = new EventWithMetadata($eventWithMetadata, []);
 
-        $this->assertSame($this->mockEvent, $nestedEventWithMetadata->getEvent());
+        self::assertSame($this->mockEvent, EventDecoratorUtilities::extractUndecoratedEvent($nestedEventWithMetadata));
```

Comment:
Is there a reason why this `EventDecoratorUtilities::extractUndecoratedEvent()` is only used here instead of the `->getEvent()` accessor?

Reply:
`getEvent()` didn't work recursively so Sebastian introduced the DecoratorUtilities with !202 - I'm not a big fan tbh but didn't know how to solve it otherwise

Reply:
Ah, okay. The recursive nature of that method wasn't obvious to me.